exonum-java-binding/cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java

```diff
@@ -92,6 +106,72 @@ public void createPublicApiHandlers(Node node, Router router) {
     });
   }
 
+  @Override
+  @Transaction(CREATE_WALLET_TX_ID)
+  public void createWallet(TxMessageProtos.CreateWalletTx arguments, TransactionContext context)
+      throws TransactionExecutionException {
+    PublicKey ownerPublicKey = context.getAuthorPk();
+
+    CryptocurrencySchema schema =
+        new CryptocurrencySchema(context.getFork(), context.getServiceName());
+    MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+
+    checkExecution(!wallets.containsKey(ownerPublicKey), WALLET_ALREADY_EXISTS.errorCode);
```

Comment:
I suppose for these checks that were moved from transaction constructors their own error codes should be introduced (and tested) - for both cryptocurrency demo and QA service.